### PR TITLE
Bug fix and added the up down for text size into visdat dialog

### LINKED ISSUE #6637: Small bug to fix on visdat dialogue, plus an improvement
[data-es-2009.zip](https://github.com/africanmathsinitiative/R-Instat/files/6870776/data-es-2009.zip)

Here is a file that can be used for the Prepare > Check Data > Visdat dialogue

It is useful to draw a sample.  I suggest 0.1.  Here is the dialogue - slightly different data

image

And here is the result:
image

Here is (roughly) the code generated - though on a slightly different file:

```
# Code generated by the dialog, Visualise Data

es_2009_2019 <- data_book$get_data_frame(data_name="es_2009_2019")

last_graph <- visdat::vis_dat(es_2009_2019 %>% dplyr::filter(as.logical(rbinom(n=nrow(es_2009_2019), size=1, prob=0.01))), sort_type=FALSE, palette="default", warn_large_data=TRUE, large_data_size=5900000) 
data_book$add_graph(graph_name="last_graph", graph=last_graph, data_name="es_2009_2019")
data_book$get_graphs(data_name="es_2009_2019", graph_name="last_graph")
rm(list=c("last_graph", "es_2009_2019"))

```
It works fine the first time.  Here is (roughly) the graph:

image

a) There is one problem to fix.  When you return to the dialogue and run again, then it gives an error.  When you examine the code the second line is now this:

`last_graph <- visdat::vis_dat(dplyr::filter(as.logical(rbinom(n=nrow(es_2009_2019), size=1, prob=0.01))), sort_type=TRUE, palette="default", warn_large_data=TRUE, large_data_size=5900000)

When you compare with the line above you can see it has now omitted this part: `es_2009_2019 %>% `
`
b) It would occasionally be good to be able to control the size of the x-text.  Here it is too large (with 130 variables) to be able to read clearly.  Now the result is a ggplot2 object, so we can add to it.  So there could be a checkbox with something like "Adjust size of variable names.  Default is unchecked.  If checked, there is an up-down.  I am not sure what the default is in the current code. (Try experimenting a bit, or look in the vid_dat function.  The up-down could be (perhaps) from 1 in steps of 0.1 to 15?  Then add the code if used as follows+
`+ ggplot2::theme(axis.text.x=ggplot2::element_text(size=5.0))`

### PR DESCRIPTION
**Pull request by N-thony**
Fixes #6637
This fixes the error when re-running the R code on reopening of the dialog and added the up-down text size option.
@africanmathsinitiative/developers @rdstern this is ready for review. Thanks.

### CODE REVIEW

**Comment by shadrackkibet** [comment]
@Ivanluv could you please review?

File: instat/dlgVisualizeData.vb [hunk 1]
```diff
@@ -92,6 +95,17 @@ Public Class dlgVisualizeData
         ucrNudSamplingFunction.Minimum = 0.01
         ucrNudSamplingFunction.Maximum = 1
 
+
+        ucrChkAdjustSize.SetText("Adjust size of variable names:")
+        ucrChkAdjustSize.AddToLinkedControls(ucrNudAdjustSize, {True}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True, bNewLinkedChangeToDefaultState:=True, objNewDefaultState:=1, bNewLinkedChangeParameterValue:=True)
```
Comment (Ivanluv) [inline instat/dlgVisualizeData.vb:100]: I dont think you need  `bNewLinkedChangeParameter ` parameter in this linking

File: instat/dlgVisualizeData.vb [hunk 2]
```diff
@@ -297,7 +323,15 @@ Public Class dlgVisualizeData
         End If
     End Sub
 
+    Private Sub ucrChkAdjustSize_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrChkAdjustSize.ControlValueChanged
+        clsBaseOperator.RemoveParameterByName("right")
```
Comment (Ivanluv) [inline instat/dlgVisualizeData.vb:327]: could have these in an else statement within the if statement below?

File: instat/dlgVisualizeData.vb [hunk 3]
```diff
@@ -297,7 +323,15 @@ Public Class dlgVisualizeData
...
+        If ucrChkAdjustSize.Checked Then
+            clsBaseOperator.AddParameter("right", clsRFunctionParameter:=clsThemeFunction, iPosition:=1)
+        End If
+    End Sub
+
     Private Sub ucrCore_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverVisualizeData.ControlContentsChanged, ucrSelectorVisualizeData.ControlContentsChanged, ucrPnlSelectData.ControlContentsChanged, ucrSaveGraph.ControlContentsChanged, ucrNudMaximumSize.ControlValueChanged, ucrNudSamplingFunction.ControlValueChanged
         TestOkEnabled()
     End Sub
+
```
Comment (Ivanluv) [inline instat/dlgVisualizeData.vb:336]: remove space

File: instat/dlgVisualizeData.vb (hunk 2 above)
Comment (N-thony) [reply instat/dlgVisualizeData.vb:327]: I think it is better rather than having additional lines with else.

**Comment by N-thony** [comment]
@Ivanluv I have resolved your comments above. Please have a look.

**Comment by Ivanluv** [comment]
@shadrackkibet  this is okay from my end

File: instat/dlgVisualizeData.vb [hunk 4]
```diff
@@ -92,6 +95,17 @@ Public Class dlgVisualizeData
...
         ucrNudSamplingFunction.Maximum = 1
 
+
+        ucrChkAdjustSize.SetText("Adjust size of variable names:")
+        ucrChkAdjustSize.AddToLinkedControls(ucrNudAdjustSize, {True}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True, bNewLinkedChangeToDefaultState:=True, objNewDefaultState:=1)
+        ucrNudAdjustSize.SetParameter(New RParameter("size", 0))
+        ucrNudAdjustSize.DecimalPlaces = 1
+        ucrNudAdjustSize.Increment = 0.1
+        ucrNudAdjustSize.Minimum = 0.1
+        ucrNudAdjustSize.Maximum = 15
```
Comment (shadrackkibet) [inline instat/dlgVisualizeData.vb:105]: Use `.SetMinMax()`?

File: instat/dlgVisualizeData.vb [hunk 5]
```diff
@@ -92,6 +95,17 @@ Public Class dlgVisualizeData
         ucrNudSamplingFunction.Minimum = 0.01
         ucrNudSamplingFunction.Maximum = 1
```
Comment (shadrackkibet) [inline instat/dlgVisualizeData.vb:96]: Alos o here

**Review by shadrackkibet — CHANGES_REQUESTED** [review]
Only one minor change.

**Comment by shadrackkibet** [comment]
@rdstern could you please check that this fixes the bug reported?

**Review by rdstern — CHANGES_REQUESTED** [review]
@N-thony this looks good.  Could you just make the default size 5 rather than 1.  Then I'll approve quickly.

**Comment by N-thony** [comment]
> @N-thony this looks good. Could you just make the default size 5 rather than 1. Then I'll approve quickly.

@rdstern I have  changed the default sized. Please have a look. Thanks.

**Review by rdstern — CHANGES_REQUESTED** [review]
@N-thony that seems fixed.  But the dialogue is now too short - so we don't see all of the bottom controls

**Comment by N-thony** [comment]
@rdstern sorry for this. I have corrected the size. Please have a look. Thanks.

File: instat/dlgVisualizeData.vb (hunk 2 above)
Comment (shadrackkibet) [reply instat/dlgVisualizeData.vb:323]: It is logical to have this as an else. Therefore I agree with @Ivanluv suggestion.

**Comment by shadrackkibet** [comment]
You added a control therefore make sure that the Tab order is sorted out.

**Comment by lloyddewit** [comment]
@N-thony I understand that you've resolved all the comments, is that correct?
There's a merge conflict, will you resolve by making a new PR and closing this PR?

**Comment by lloyddewit** [comment]
Replaced by PR #6666

**Pull request closed without merging**